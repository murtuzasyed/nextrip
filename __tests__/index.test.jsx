import { render, screen } from '@testing-library/react'
// import userEvent from '@testing-library/user-event';
import Home from '../pages/index';


describe('Home', () => {
  const routes=[{
    label: "Mock Option",
    value: "mockOption"
  }];
  it("renders correctly", () => {
    const { container } = render(<Home routes={routes}/>)
    expect(container).toMatchSnapshot()
  })
  it('renders a heading', () => {
  
    render(<Home routes={routes}/>)

    const heading = screen.getByRole('heading', {
      name: /Next Trip/i,
    })

    expect(heading).toBeInTheDocument()
  });
  it("displays only one option", () => {
    render(<Home routes={routes}/>)
    expect(screen.getAllByRole('option').length).toBe(1);
  });
})