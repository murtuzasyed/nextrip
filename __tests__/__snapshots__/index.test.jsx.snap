// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders correctly 1`] = `
<div>
  <div
    class="container"
  >
    <h1>
      Next Trip
    </h1>
    <div
      class="nextrip-container"
    >
      <div
        class="nextrip-selectgroup"
      >
        <select
          class="custom-select"
          id="routes"
          name="routes"
        >
          <option
            value="mockOption"
          >
            Mock Option
          </option>
        </select>
      </div>
    </div>
  </div>
</div>
`;
